cdap-unit-test/src/test/java/co/cask/cdap/test/app/WorkflowAppWithLocalDatasets.java

```diff
@@ -68,26 +84,108 @@ protected void configure() {
       setName("WorkflowWithLocalDatasets");
       setDescription("Workflow program with local datasets.");
       createLocalDataset(WORDCOUNT_DATASET, KeyValueTable.class);
+      createLocalDataset(CSV_FILESET_DATASET, FileSet.class, FileSetProperties.builder()
+        .setInputFormat(TextInputFormat.class)
+        .setOutputFormat(TextOutputFormat.class)
+        .build());
+      addAction(new LocalDatasetWriter());
+      addSpark("JavaSparkCSVToSpaceConverter");
       addMapReduce("WordCount");
       addAction(new LocalDatasetReader());
     }
   }
 
   /**
+   * Custom action writing to the local file set dataset.
+   */
+  public static class LocalDatasetWriter extends AbstractWorkflowAction {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LocalDatasetWriter.class);
+
+    @Override
+    public void run() {
+      String inputPath = getContext().getRuntimeArguments().get("input.path");
+      FileSet fileSetDataset = getContext().getDataset(CSV_FILESET_DATASET);
+      Location inputLocation = fileSetDataset.getLocation(inputPath);
+
+      try {
+        try (PrintWriter writer = new PrintWriter(inputLocation.getOutputStream())) {
+          writer.write("this,text,has");
+          writer.println();
+          writer.write("two,words,text,inside");
+        }
+      } catch (Throwable t) {
+        LOG.info ("Exception occurred while running custom action ", t);
+      }
+    }
+  }
+
+  /**
+   * Spark program to convert comma separated file into space separated file.
+   */
+  public static final class JavaSparkCSVToSpaceConverter extends AbstractSpark {
+    @Override
+    protected void configure() {
+      setName("JavaSparkCSVToSpaceConverter");
+      setMainClass(SparkCSVToSpaceProgram.class);
+      setDriverResources(new Resources(2048));
+      setExecutorResources(new Resources(2048));
+    }
+  }
+
+
+  /**
+   * Main class for the Spark program to convert comma separated file into space separated file.
+   */
+  public static final class SparkCSVToSpaceProgram implements JavaSparkProgram {
+
+    @Override
+    public void run(SparkContext context) throws Exception {
+      Map<String, String> fileSetArgs = new HashMap<>();
+      FileSetArguments.addInputPath(fileSetArgs, context.getRuntimeArguments().get("input.path"));
+      JavaPairRDD<LongWritable, Text> input = context.readFromDataset(CSV_FILESET_DATASET, LongWritable.class,
+                                                                      Text.class, fileSetArgs);
+
+      JavaPairRDD<LongWritable, String> converted = input.mapToPair(new PairFunction<Tuple2<LongWritable, Text>,
```

Comment:
Since you don't need the key, the code would be simpler if you do:

input.values().map(new Function<Text, String>() { ... });